src/pages/settings/gateways/common/components/Tabs.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { useAccentColor } from 'common/hooks/useAccentColor';
+import { Dispatch, SetStateAction } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  className?: string;
+  tabs: Tab[];
+  tabsActivity: PaymentTabsActivity;
+  setTabsActivity: Dispatch<SetStateAction<PaymentTabsActivity>>;
+  activeTabKey: string;
+}
+
+export interface Tab {
+  name: string;
+}
+
+export interface PaymentTabsActivity {
+  overview: boolean;
+  credentials: boolean;
+  settings: boolean;
+  required_fields: boolean;
+  limits_and_fees: boolean;
+}
+
+export function Tabs(props: Props) {
```

Comment:
The reason for custom cards for Payment Gateways is the data sharing logic implemented on the Gatways component. The layout of the tabs is exactly the same, but the main difference is that the tabs are based on breadcrumbs. So when you change the card as well, the breadcrumb will change. I think our main Tabs component is not configurable in this case.

Reply:
Hi, 

I am not sure what you mean. Any chance you can record a quick demo explaining why do we need custom tabs for this?

Reply:
> Hi,
> 
> I am not sure what you mean. Any chance you can record a quick demo explaining why do we need custom tabs for this?

Hi Ben, it's pretty simple, I'll use a Company Details cards as an example. There we have several tabs/components. All data in components is managed by Redux (handleChange/useSelector) and that's why we can navigate to a URL and render a component, the data will always be in Redux and available for display or use. Here we need to pass the selected gateway on the Overview tab through all components. If we navigate to a different URL, we will never know which Gateway is selected. The solution to using the already created Tabs component is to switch the logic to Redux. If that's okay for you, I can work on the switch. I'm pretty sure you'll understand what I'm talking about, but if you don't I'll be recording a demo tomorrow, I'm not in a position to record it right now.

![image](https://user-images.githubusercontent.com/51542191/204165987-4dd8b656-50ef-4ffe-83f6-d3d97898a79a.png)